encoding/mkv.md

```diff
@@ -0,0 +1,505 @@
+# Mastroka Container Tricks
+
+*Disclaimer: this page assumes some basic knowledge of the Mastroka container,
+Python, and VapourSynth.*
+
+
+## Background and Preparation
+
+The Mastroka Media Container is an open source,
+open standard multimedia container
+that is heavily used in the fansubbing scene.
+Mastroka video, or `.mkv` files
+are the standard for soft-subbed releases
+as they allow for multiple subtitles tracks and chapters
+which are a somewhat unique feature to the container.
+The MPEG-4 Part 14 or MP4 is another container format
+but it is limited in its usefulness for fansubbing.
+
+The following steps will require
+VapourSynth,
+[MKVToolNix GUI + CLI][toolnix],
+and [FFmpeg][] to be installed
+(or readily available as binaries).
+A [Mastroka-friendly player][mk-playback], preferably [MPC-HC][] or [mpv][],
+will also be required for testing the files.
+
+The examples will be using pre-made subtitle tracks,
+a BDMV source,
+[VSEdit](preparation.md#the-editor),
+an [audio-cutting Python script][acsuite.py],
+and a multitude of utility scripts/functions to help speed up the process.
+
+What we will be doing:
+1. Extracting audio tracks from BDMV .m2ts files
+1. Trimming audio to match our VapourSynth video trims
+1. Creating mkv files from individual tracks
+1. Creating chapters for said files
+1. Creating virtual timelines with ordered chapters
+1. Using multiple audio codecs in the same 'apparent file'
+1. Using editions to change between NCOP and OP
+1. Creating a playall file with ordered chapters
+
+[toolnix]: https://mkvtoolnix.download/downloads.html
+[FFmpeg]: https://ffmpeg.zeranoe.com/builds/
+[mk-playback]: https://github.com/hubblec4/Matroska-Playback/blob/e48149b500ce80c38ebc07ccf8f74f8af45e89b2/src/PlayerOverview.md
+[MPC-HC]: https://github.com/clsid2/mpc-hc/releases
+[mpv]: https://mpv.io/installation/
+[acsuite.py]: https://github.com/OrangeChannel/acsuite
+
+
+### Starting with a clean work environment
+
+The following steps should be done from within a folder containing
+all of the files (episodes) re-named to some simple format like `ep##.m2ts`,
+as having multiple `00004.m2ts` files won't be helpful.
+The example scripts provided will assume this workflow.
+
+
+## Extracting the audio
+
+If the source files you are working with are not .m2ts,
+re-encoding your audio into PCM / WAV is *still* recommended
+as slicing some lossless or lossy codecs is not precise.
```

Comment:
What about FLAC here? Are there lossless codecs that don't decode or losslessly convert to WAV/PCM? All others should be easily spliceable.

Reply:
I'm going to leave the testing up to someone else for the time being. The whole 'not starting with a BDMV source' part of this page is gonna be a TODO for some time.